### PR TITLE
port `DocumentKey` to non-container types of `Model/*`

### PR DESCRIPTION
**Pull request by zxu123**
### Discussion
Part of the C++ migration.

### Testing
unit and integration tests pass.

### API Changes
n/a

### CODE REVIEW

**Comment by zxu123** [comment]
Please ignore the historical conversation; this branch is branched from other pending branch before that other branch merged. Now the other branch is merged into master, which is merged into this branch; the `files changed` is clean.

The relevant commit starts at:
"using DocumentKey in model except for the container types `FSTDocumen…  …" b64c1c1

File: Firestore/core/test/firebase/firestore/testutil/testutil.h [hunk 1]
```diff
@@ -27,6 +27,10 @@ namespace testutil {
 
 // Below are convenience methods for creating instances for tests.
 
+inline model::DocumentKey DocKey(absl::string_view path) {
```
Comment (wilhuff) [inline Firestore/core/test/firebase/firestore/testutil/testutil.h]: This function is just called key on Android and JS; let's call it that here too.

File: Firestore/Source/Core/FSTView.mm [hunk 2]
```diff
@@ -216,8 +216,8 @@ - (FSTViewDocumentChanges *)computeChangesWithDocuments:(FSTMaybeDocumentDiction
       newDoc = (FSTDocument *)maybeNewDoc;
     }
     if (newDoc) {
-      FSTAssert([key isEqual:newDoc.key], @"Mismatching key in document changes: %@ != %@", key,
-                newDoc.key);
+      FSTAssert([key isEqual:newDoc.key], @"Mismatching key in document changes: %@ != %s", key,
+                newDoc.key.path().CanonicalString().c_str());
```
Comment (wilhuff) [inline Firestore/Source/Core/FSTView.mm]: Should we add a ToString() on DocumentKey to cut down on this clutter in the log lines?

File: Firestore/core/test/firebase/firestore/testutil/testutil.h (hunk 1 above)
Comment (zxu123) [reply Firestore/core/test/firebase/firestore/testutil/testutil.h]: done

File: Firestore/Source/Core/FSTView.mm (hunk 2 above)
Comment (zxu123) [reply Firestore/Source/Core/FSTView.mm]: done

**Pull request merged**